php_mongo.h

```diff
@@ -620,6 +620,7 @@ extern zend_module_entry mongo_module_entry;
  * 18: Trying to get more, but cannot find server
  * 19: max number of retries exhausted, couldn't send query
  * 20: something exceptional has happened, and the cursor is now dead
+ * 21: Reading data for type %02x would exceed buffer for key "%s"
  * 21: invalid string length for key "%s"
```

Comment:
This message/code was already being used by the `CHECK_BUFFER_LEN` macro, so I'm just documenting existing behavior.

I did introduce a new message "Reading key name for type %02x would exceed buffer" and used the same code, as it's logically very similar to the "Reading data" error except for the fact that we couldn't even read the field name string. I didn't think it was worth adding a new code for that, and was on the fence about documenting the variation in the error message. Open to any input on that, though.

I do think re-using the code makes sense, but I'm not really against adding the message here, too.